### PR TITLE
Feat/chart mode

### PR DESCRIPTION
**Pull request by mayman99**

### CODE REVIEW

File: src/panel/chart.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import * as React from 'react';
+import { PureComponent } from 'react';
+import { ResultTableStore } from './resultTableStore';
+import * as d3 from 'd3';
+
+export interface ChartProps<G> {
+    store: ResultTableStore<G>;
+}
+@observer export class Chart<G> extends PureComponent<ChartProps<G>> {
+    private ref!: SVGSVGElement;
```
Comment (audrow) [inline src/panel/chart.tsx:13]: Why is there an exclamation here?

File: src/panel/chart.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import { PureComponent } from 'react';
+import { ResultTableStore } from './resultTableStore';
+import * as d3 from 'd3';
```
Comment (audrow) [inline src/panel/chart.tsx:7]: Not saying anything needs to change, but is there a reason why we didn't use [chart js](https://www.chartjs.org/), or any other simpler plotting library?

File: src/panel/chart.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    componentDidMount() {
+        // activate
+        this.pieChart();
+    }
+
+    render() {
+        return (<div className="svg">
+            <svg className="container" ref={(ref: SVGSVGElement) => this.ref = ref}></svg>
```
Comment (audrow) [inline src/panel/chart.tsx:110]: This is a cool way to use D3 inside of React.

**Review by audrow — APPROVED** [review]
A few questions, but looks good to me!

File: src/panel/chart.tsx (hunk 1 above)
Comment (mayman99) [reply src/panel/chart.tsx:13]: tells the TypeScript compiler that a value typed as optional cannot be null or undefined

File: src/panel/chart.tsx (hunk 2 above)
Comment (mayman99) [reply src/panel/chart.tsx:7]: I didn't know about chart js

**Pull request merged**